### PR TITLE
(Un)set role for multiple users/groups

### PR DESCRIPTION
**Pull request by Vojtech-Sassmann**
* Problem: we need to be able to set assing a role to multiple
users/groups so it can be done with only one call e.g. from GUI.

* Created new variations of methods for (un)setings roles to multiple
users or groups.

### CODE REVIEW

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AuthzResolverMethod.java [hunk 1]
```diff
@@ -295,12 +355,40 @@ public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
...
+	/*#
+	 * Unset role for users and complementaryObject.
+	 *
+	 * If complementary object is wrong for the role, throw an exception.
+	 * For role "perunadmin" ignore complementary object.
+	 *
+	 * IMPORTANT: Refresh authz only if user in session is affected.
+	 *
+	 * @param role String Role which will be unset for given users ( FACILITYADMIN | GROUPADMIN | PERUNADMIN | RESOURCEADMIN | RESOURCESELFSERVICE | SPONSOR | TOPGROUPCREATOR | VOADMIN | VOOBSERVER | PERUNOBSERVER | SECURITYADMIN | CABINETADMIN )
+	 * @param users List<int> <code>ids</code> of users for which is the role set
```
Comment (zlamalp) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AuthzResolverMethod.java]: Docs parser requires either ``int[]`` or ``List<Integer>`` as a type.

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AuthzResolverMethod.java [hunk 2]
```diff
@@ -295,12 +355,40 @@ public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
...
+	/*#
+	 * Unset role for authorizedGroups and complementaryObject.
+	 *
+	 * If complementary object is wrong for the role, throw an exception.
+	 * For role "perunadmin" ignore complementary object.
+	 *
+	 * IMPORTANT: Refresh authz only if user in session is affected.
+	 *
+	 * @param role String Role which will be set for given groups ( FACILITYADMIN | GROUPADMIN | PERUNADMIN | RESOURCEADMIN | RESOURCESELFSERVICE | SPONSOR | TOPGROUPCREATOR | VOADMIN | VOOBSERVER | PERUNOBSERVER | SECURITYADMIN | CABINETADMIN )
+	 * @param authorizedGroups List<int> <code>ids</code> of groups for which is the role set
```
Comment (zlamalp) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AuthzResolverMethod.java]: Docs parser requires either ``int[]`` or ``List<Integer>``.

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AuthzResolverMethod.java [hunk 3]
```diff
@@ -188,12 +190,40 @@ public AuthzRoles call(ApiCaller ac, Deserializer parms) throws PerunException {
...
+	/*#
+	 * Set role for authorizedGroups and complementaryObject.
+	 *
+	 * If complementary object is wrong for the role, throw an exception.
+	 * For role "perunadmin" ignore complementary object.
+	 *
+	 * IMPORTANT: Refresh authz only if user in session is affected.
+	 *
+	 * @param role String Role which will be set for given groups ( FACILITYADMIN | GROUPADMIN | PERUNADMIN | RESOURCEADMIN | RESOURCESELFSERVICE | SPONSOR | TOPGROUPCREATOR | VOADMIN | VOOBSERVER | PERUNOBSERVER | SECURITYADMIN | CABINETADMIN )
+	 * @param authorizedGroups List<int> <code>ids</code> of groups for which is the role set
```
Comment (zlamalp) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AuthzResolverMethod.java]: Docs parser requires either ``int[]`` or ``List<Integer>``.

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AuthzResolverMethod.java [hunk 4]
```diff
@@ -188,12 +190,40 @@ public AuthzRoles call(ApiCaller ac, Deserializer parms) throws PerunException {
...
+	/*#
+	 * Set role for users and complementaryObject.
+	 *
+	 * If complementary object is wrong for the role, throw an exception.
+	 * For role "perunadmin" ignore complementary object.
+	 *
+	 * IMPORTANT: Refresh authz only if user in session is affected.
+	 *
+	 * @param role String Role which will be set for given users ( FACILITYADMIN | GROUPADMIN | PERUNADMIN | RESOURCEADMIN | RESOURCESELFSERVICE | SPONSOR | TOPGROUPCREATOR | VOADMIN | VOOBSERVER | PERUNOBSERVER | SECURITYADMIN | CABINETADMIN )
+	 * @param users List<int> <code>ids</code> of users for which is the role set
```
Comment (zlamalp) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/AuthzResolverMethod.java]: Docs parser requires either ``int[]`` or ``List<Integer>``.

**Comment by Vojtech-Sassmann** [comment]
Docs fixed.

**Pull request merged**